aptos-move/e2e-move-tests/tests/rotate_auth_key.rs

```diff
@@ -88,3 +88,107 @@ fn rotate_auth_key() {
+    // and the new public key
+    let rotation_proof = RotationProofChallenge {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationProofChallenge"),
+        sequence_number: 10,
+        originator: *account1.address(),
+        current_auth_key: AccountAddress::from_bytes(&account1.auth_key()).unwrap(),
+        new_public_key: new_public_key.to_bytes().to_vec(),
+    };
+
+    let rotation_msg = bcs::to_bytes(&rotation_proof);
+
+    // sign the struct using both the current private key and the next private key
+    let rotation_proof_signed_by_current_private_key = account1
+        .privkey
+        .sign_arbitrary_message(&rotation_msg.clone().unwrap());
+    let rotation_proof_signed_by_new_private_key =
+        new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &account1,
+        aptos_stdlib::account_rotate_authentication_key_ed25519(
+            rotation_proof_signed_by_current_private_key
+                .to_bytes()
+                .to_vec(),
+            rotation_proof_signed_by_new_private_key.to_bytes().to_vec(),
+            account1.pubkey.to_bytes().to_vec(),
+            new_public_key.to_bytes().to_vec(),
+        )
+    ));
+
+    // get the address redirection table
+    let originating_address_handle = harness
+        .read_resource::<TableHandle>(
+            &CORE_CODE_ADDRESS,
+            parse_struct_tag("0x1::account::OriginatingAddress").unwrap(),
+        )
+        .unwrap();
+    let state_key = &StateKey::table_item(
+        originating_address_handle,
+        AccountAddress::from_bytes(new_auth_key).unwrap().to_vec(),
+    );
+    // verify that the value in the address redirection table is expected
+    let result = harness.read_state_value(state_key).unwrap();
+    assert_eq!(result, address.to_vec());
+
+    let new_private_key2 = Ed25519PrivateKey::generate_for_testing();
+    let new_public_key2 = new_private_key2.public_key();
+    let new_auth_key2 = AuthenticationKey::ed25519(&new_public_key2);
+
+    let rotation_proof2 = RotationProofChallenge {
+        account_address: CORE_CODE_ADDRESS,
+        module_name: String::from("account"),
+        struct_name: String::from("RotationProofChallenge"),
+        sequence_number: 11,
+        originator: *account1.address(),
+        current_auth_key: AccountAddress::from_bytes(account2.auth_key()).unwrap(),
+        new_public_key: new_public_key2.to_bytes().to_vec(),
+    };
+
+    let rotation_msg2  = bcs::to_bytes(&rotation_proof2);
+
+    let rotation_proof_signed_by_current_private_key2 = account1.privkey.sign_arbitrary_message(&rotation_msg2.clone().unwrap());
+    let rotation_proof_signed_by_new_private_key2 = new_private_key2.sign_arbitrary_message(&rotation_msg2.unwrap());
+
+    assert_success!(harness.run_transaction_payload(
+        &account1,
```

Comment:
Please try rotate with &account2 and see what happens. 
The scenario is "the owner of account1 rotates to account2's auth_key" 
Now, owner use account2's key and use it for further rotation